storage/factory/pruningStorerFactory.go

```diff
@@ -40,7 +40,7 @@ func NewStorageServiceFactory(
 	if config == nil {
 		return nil, storage.ErrNilConfig
 	}
-	if config.StoragePruning.NumEpochsToKeep < minimumNumberOfEpochsToKeep && !config.StoragePruning.FullArchive {
+	if config.StoragePruning.NumEpochsToKeep < minimumNumberOfEpochsToKeep && !config.StoragePruning.CleanOldEpochsData {
```

Comment:
why is cleanOldEpochsData relevant here?

Reply:
If the node doesn't remove old epochs, then there is no use for the `NumEpochsToKeep` so the check shouldn't be blocking. I can remove the condition if needed. Also, thanks for finding this because I have to change the boolean condition

Reply:
ok then you can keep the check, but remove negation